### PR TITLE
Templated munkres

### LINKED ISSUE #18: Adapters doesn't allow correct convert non square matrices
This bug was found by @kaajo . But unfortunately proposed by @kaajo [changes](https://github.com/Gluttton/munkres-cpp/pull/1) are not complete and correct and require fixes.

### PR DESCRIPTION
**Pull request by kaajo**
I think its good idea to have munkres as template

Added:
- abstract Adapter class (all adapters are derived)
- cmake for all adapters (installed in <include_path>/munkres/adapters/)
- cmake options for installing adapters

Removed:
- raw_2d_array adapter

Should fix #18

### CODE REVIEW

File: src/munkres.h [hunk 1]
```diff
@@ -23,29 +24,441 @@
...
+     * (extra 0 values are replaced with -1)
+     *
+     */
+    void solve(Matrix<Data> &m) {
+        const size_t rows = m.rows(),
+                columns = m.columns(),
+                size = std::max(rows, columns);
+
+#ifdef DEBUG
+        std::cout << "Munkres input: " m << std::endl;
```
Comment (saebyn) [inline src/munkres.h]: Missing << after string.

File: tests/adapters/raw_2d_arraytest.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #include <gtest/gtest.h>
-#include "adapters/raw_2d_array.h"
+#include "adapters/"
```
Comment (saebyn) [inline tests/adapters/raw_2d_arraytest.cpp]: Missing something here?

Comment (kaajo) [inline tests/adapters/raw_2d_arraytest.cpp]: i forgot to remove  tests for raw_2d_array adapter, fix in next commit

**Comment by saebyn** [comment]
Looking good. I'll make some time this weekend to look at this in more depth. Thank you for contributing!

**Pull request merged**